### ISSUE TITLE
Failed to resolve 'Device:Startup' component from 'legacy' ST DFP packs

### ISSUE THREAD

**Issue by chentang16** [ARM-software/vscode-cmsis-csolution#111]
Type: Bug Report

**Describe the bug**
- OS and Version: Windows 11
- VS Code Version: 1.95.1
- Extension Version: 1.40.0
- Target Device: STM32F303RET (NUCLEO-F303RE)

When creating a new STM32F303RET (NUCLEO-F303RE) csolution project from the latest STM32F3xx_DFP pack v2.2.2 (not clayer based), Software Components in VS Code failed to resolve 'Device:Startup' component, no matter if the "standalone" or the "STM32CUBEMX" variant is selected.

Build error output:
error csolution: no component was found with identifier 'Device:Startup'
error csolution: processing context 'test.Debug+STM32F303RETx' failed
Process failed with exit code 1
Process failed with exit code 1

The same new project creation process done in MDK v5 µVision IDE works fine as expected, as shown in the attached screenshot.
![2024-11-04_14-44-44](https://github.com/user-attachments/assets/13d228c2-5241-4932-87ed-19adc2361c5b)

**Comment by ReinhardKeil** [comment]
Could you add the csolution project as ZIP file?

**Comment by chentang16** [comment]
Here is my csolution project
[Blank_solution.zip](https://github.com/user-attachments/files/17620006/Blank_solution.zip)

**Comment by jkrech** [comment]
The selected "Template Project" is the "Blank Solution" which already has the component "Device:Startup" selected in the "Project.cproject.yml" file. 
However the STM32F3xx_DFP describes the "Device:Startup" component in two Cbundles "Standalone" and "STM32CubeMX" and because present Device:Startup has no bundle at all, the Component displays in the Variant column "None" , "Standalone", "STM32CubeMX"

You first need to unselect the "Device:Startup" component that cannot be resolved.
Afterwards you need to either select "Standalone" or "STM32CubeMX" from the variant column and select the "Device:Startup" component from the respective bundle.
Now you should see e.g. 
```
    - component: Keil::Device&Standalone:Startup
 
```
in the cproject.yml file"Device Startup"

**Comment by ReinhardKeil** [comment]
We need to improve visiblity in the UX

**Comment by chentang16** [comment]
> The selected "Template Project" is the "Blank Solution" which already has the component "Device:Startup" selected in the "Project.cproject.yml" file. However the STM32F3xx_DFP describes the "Device:Startup" component in two Cbundles "Standalone" and "STM32CubeMX" and because present Device:Startup has no bundle at all, the Component displays in the Variant column "None" , "Standalone", "STM32CubeMX"
> 
> You first need to unselect the "Device:Startup" component that cannot be resolved. Afterwards you need to either select "Standalone" or "STM32CubeMX" from the variant column and select the "Device:Startup" component from the respective bundle. Now you should see e.g.
> 
> ```
>     - component: Keil::Device&Standalone:Startup
> ```
> 
> in the cproject.yml file"Device Startup"
> 
> A problem of the component window is, that it allows to concurrently select components from different bundles without raising a warning/error. Also `cbuild` does not complain about two Device:Startup components:
> 
> ```
>   components:
>     - component: Keil::Device&Standalone:Startup
>     - component: Keil::Device&STM32CubeMX:Startup
> ```

Yes, thx! This indeed works. However, the first step "You first need to unselect the "Device:Startup" component that cannot be resolved" I have to repeat two/three times to properly unselect it. It seems to me each time of click reflects one validator error shown in red in my screenshot above.

**Comment by jkrech** [comment]
CMSIS Solution Extension 1.66

<img width="926" height="298" alt="Image" src="https://github.com/user-attachments/assets/9da0c188-532a-4da4-b035-aa40656fe698" />

unselect "Startup" component and `Save`

<img width="929" height="110" alt="Image" src="https://github.com/user-attachments/assets/7a5ef62a-43ff-4b57-8106-4e02c52ddfec" />

select "Startup" from either bundle.
